src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_GetModelUpdates.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineViewModel
+    {
+        /// <summary>
+        /// Queue that uses the language-server-protocol to get document symbol information.
+        /// This queue can return null if it is called before and LSP server is registered for our document.
+        /// </summary>
+        private readonly AsyncBatchingWorkQueue<DocumentSymbolRequestInfo, DocumentSymbolDataModel?> _documentSymbolQueue;
+
+        private async ValueTask<DocumentSymbolDataModel?> GetDocumentSymbolAsync(ImmutableSegmentedList<DocumentSymbolRequestInfo> documentSymbolRequestInfos, CancellationToken cancellationToken)
+        {
+            var (textBuffer, filePath) = documentSymbolRequestInfos.Last();
```

Comment:
if this is all that the is in the infos, a simpler approach would be to just have this be an AsyncBatchingWorkQueue with no arg type.  And just have the code determine the textbuffer/filepath *here* when it executes.